integration/dockerfiles/Dockerfile_test_user_home

```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Google, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM debian:9.11
+
+# default values for the root user.
+RUN touch $HOME/hello
+
+# testuser1 with the default home created by useradd.
+
+RUN groupadd testgroup && \
+    useradd --create-home --gid testgroup alice && \
+    useradd --create-home --uid 1111 --home-dir /home/john --gid testgroup bob
+
+USER alice:testgroup
+
+RUN touch $HOME/hello
+
+USER bob:testgroup
+
+RUN touch $HOME/hello
+
+USER root
+
+USER 1111
```

Comment:
New issue, it seems that Docker produces `USER 1111:1000`, Kaniko produces `USER 1111`

Reply:
I just ran your PR on saw, kaniko add the following metadata

```

{"architecture":"amd64","container":"92d26d114b4443b842cef348a7fc3d0c7dfe4480704f8d2cf8a837b352bb490a","created":"2020-03-04T17:57:11.614785227Z","docker_version":"18.09.7","history":[{"created":"2020-02-01T17:23:41.468764783Z","created_by":"/bin/sh -c #(nop) ADD file:8a9218592e5d736a05a1821a6dd38b205cdd8197c26a5aa33f6fc22fbfaa1c4d in / "},{"created":"2020-02-01T17:23:41.779347941Z","created_by":"/bin/sh -c #(nop)  CMD [\"bash\"]","empty_layer":true},{"author":"kaniko","created":"0001-01-01T00:00:00Z","created_by":"RUN touch $HOME/hello"},{"author":"kaniko","created":"0001-01-01T00:00:00Z","created_by":"RUN groupadd testgroup \u0026\u0026     useradd --create-home --gid testgroup alice \u0026\u0026     useradd --create-home --uid 1111 --home-dir /home/john --gid testgroup bob"},{"author":"kaniko","created":"0001-01-01T00:00:00Z","created_by":"RUN touch $HOME/hello"},{"author":"kaniko","created":"0001-01-01T00:00:00Z","created_by":"RUN touch $HOME/hello"},{"author":"kaniko","created":"0001-01-01T00:00:00Z","created_by":"RUN touch ~/hello"}],"os":"linux","rootfs":{"type":"layers","diff_ids":["sha256:7948c3e5790c6df89fe48041fabd8f1c576d4bb7c869183e03b9e3873a5f33d9","sha256:5763bdc853d65871befe87dd94ff8035447c69707230ba54d7507e061f64bdcf","sha256:787f2406f0f160e50e82c870090de259b6df059878286f2952d73a3e4b6940b3","sha256:6ec367cd5b8bbdf1e8a1e87b774cb41ffa9e25cdecf44b37162758d6ec08c98b","sha256:9392344d0a95074a2bb4df978fb9bce67a6b9338d2385ccdde9040fe978e686f","sha256:9dc6c2ce42c12e21b20d8badfb3e3a9785f9b8debbe7d75fbab8af657913536d"]},"config":{"Cmd":["bash"],"Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],"Image":"sha256:ac0cc3c8385b2b16438d2f165b2461d0412339b2292436bcf8f8b5835cb0b04c","User":"1111:1000","ArgsEscaped":true}}

```
The user is set to `1111:1000`

While docker config is User is `1111`
```
{"architecture":"amd64","config":{"Hostname":"","Domainname":"","User":"1111","AttachStdin":false,"AttachStdout":false,"AttachStderr":false,"Tty":false,"OpenStdin":false,"StdinOnce":false,"Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],"Cmd":["bash"],"ArgsEscaped":true,"Image":"sha256:a3743be40385cec5e8a25e39381eca308c72b32886730da7ddd29159917f97aa","Volumes":null,"WorkingDir":"","Entrypoint":null,"OnBuild":null,"Labels":null},"container":"5667498e5ae53afb26f7196c744464212a478ce80fdc3d3b7ecda7cbd16eb2b9","container_config":{"Hostname":"","Domainname":"","User":"1111","AttachStdin":false,"AttachStdout":false,"AttachStderr":false,"Tty":false,"OpenStdin":false,"StdinOnce":false,"Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],"Cmd":["/bin/sh","-c","touch ~/hello"],"Image":"sha256:a3743be40385cec5e8a25e39381eca308c72b32886730da7ddd29159917f97aa","Volumes":null,"WorkingDir":"","Entrypoint":null,"OnBuild":null,"Labels":null},"created":"2020-03-04T18:03:27.316322906Z","docker_version":"19.03.5","history":[{"created":"2020-02-01T17:23:41.468764783Z","created_by":"/bin/sh -c #(nop) ADD file:8a9218592e5d736a05a1821a6dd38b205cdd8197c26a5aa33f6fc22fbfaa1c4d in / "},{"created":"2020-02-01T17:23:41.779347941Z","created_by":"/bin/sh -c #(nop)  CMD [\"bash\"]","empty_layer":true},{"created":"2020-03-04T18:03:23.576015951Z","created_by":"/bin/sh -c touch $HOME/hello"},{"created":"2020-03-04T18:03:24.53030722Z","created_by":"/bin/sh -c groupadd testgroup \u0026\u0026     useradd --create-home --gid testgroup alice \u0026\u0026     useradd --create-home --uid 1111 --home-dir /home/john --gid testgroup bob"},{"created":"2020-03-04T18:03:24.643632066Z","created_by":"/bin/sh -c #(nop)  USER alice:testgroup","empty_layer":true},{"created":"2020-03-04T18:03:25.439401085Z","created_by":"/bin/sh -c touch $HOME/hello"},{"created":"2020-03-04T18:03:25.575861277Z","created_by":"/bin/sh -c #(nop)  USER bob:testgroup","empty_layer":true},{"created":"2020-03-04T18:03:26.328779782Z","created_by":"/bin/sh -c touch $HOME/hello"},{"created":"2020-03-04T18:03:26.454558348Z","created_by":"/bin/sh -c #(nop)  USER root","empty_layer":true},{"created":"2020-03-04T18:03:26.587195086Z","created_by":"/bin/sh -c #(nop)  USER 1111","empty_layer":true},{"created":"2020-03-04T18:03:27.316322906Z","created_by":"/bin/sh -c touch ~/hello"}],"os":"linux","rootfs":{"type":"layers","diff_ids":["sha256:7948c3e5790c6df89fe48041fabd8f1c576d4bb7c869183e03b9e3873a5f33d9","sha256:1c9d60ae5c222d2736e408686f0c92fe0ff434d7bc0e5098ca4c06dd1825b704","sha256:f2ce2a74a9375df3927d56599c97331f49a6959014dac9ca45d09ac88d737de0","sha256:3a8bd636c26cdcd1aa4b494d29879babffa163da7acdc542b31647d721f8ad6e","sha256:e4c469eeb8b3f89e32e70a520820cd8cb3de5c71f01aa34149d06e2b6e94b8a6","sha256:63b1617af7435d6975185d3d41ca199da7ae2e5c81892cf0678d4f00f71e7c76"]}}  
````

Reply:
I created a fix here https://github.com/GoogleContainerTools/kaniko/pull/1105
Please reivew and i will merge this.

Reply:
I did something during my commute today before I get to have look to the PR you've just made. I'll give it a look tomorrow. Cheers!

Reply:
No worries. Thanks for taking this home @greut